LoopBack/LBModel.m

```diff
@@ -22,6 +21,83 @@ - (NSMutableDictionary *)_overflow;
 
 static NSDateFormatter *jsonDateFormatter = nil;
 
+@implementation NSArray (LBModelAdditions)
+
+- (NSMutableData*) convertToMutableData {
+    NSUInteger n = [self count];
+    NSMutableData *data = [NSMutableData dataWithLength: n];
+    char *p = [data mutableBytes];
+    for (int i = 0; i < [self count]; i++) {
+        *p++ = [[self objectAtIndex:i] charValue];
+    }
+    return data;
+}
+
+@end
+
+@implementation NSData (LBModelAdditions)
+
+- (NSDictionary *)convertToJSONObject {
+    NSMutableDictionary *jsonObject = [NSMutableDictionary dictionary];
+    [jsonObject setObject:@"Buffer" forKey:@"type"];
+
+    NSMutableArray *array = [NSMutableArray array];
+    [self enumerateByteRangesUsingBlock:^(const void *bytes, NSRange byteRange, BOOL *stop) {
+        for (NSInteger i = 0; i < byteRange.length; i++) {
+            [array addObject:[NSNumber numberWithChar:((char*)bytes)[i]]];
+        }
+    }];
+    [jsonObject setObject:array forKey:@"data"];
+
+    return jsonObject;
+}
+
+@end
+
+@implementation NSMutableData (LBModelAdditions)
+
++ (NSMutableData *)dataFromJSONObject:(NSDictionary *)jsonObject {
+    // jsonObject is assumed to be a Node.js Buffer object
+    if ([jsonObject count] != 2) {
+        return nil;
+    }
+    if (![(NSString *)(jsonObject[@"type"]) isEqualToString:@"Buffer"]) {
+        return nil;
+    }
+    if (![jsonObject[@"data"] isKindOfClass:[NSArray class]]) {
+        return nil;
+    }
+    NSArray *array = jsonObject[@"data"];
+    NSMutableData *data = [array convertToMutableData];
+
+    return data;
+}
+
+@end
+
+@implementation CLLocation (LBModelAdditions)
+
++ (CLLocation *)locationFromJSONObject:(NSDictionary *)jsonObject {
+    if ([jsonObject count] != 2 || jsonObject[@"lat"] == nil || jsonObject[@"lng"] == nil) {
+        return nil;
```

Comment:
I think this will be difficult to troubleshoot. Could you please add a debug log or some kind of a warning, to tell the developer that invalid data was received for a location property?